jdaviz/core/config.py

```diff
@@ -0,0 +1,98 @@
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Filename: config.py
+# Project: core
+# Author: Brian Cherinka
+# Created: Tuesday, 23rd June 2020 1:22:32 pm
+# License: BSD 3-clause "New" or "Revised" License
+# Copyright (c) 2020 Brian Cherinka
+# Last Modified: Tuesday, 23rd June 2020 1:22:32 pm
+# Modified By: Brian Cherinka
+
+#
+#  This file contains helper function related to configuration handling
+#
+
+from __future__ import print_function, division, absolute_import
+import copy
+import os
+import pathlib
+import yaml
+
+
+def read_configuration(path=None):
+    """ Loads a configuration from a YAML file
+
+    Loads application configuration settings from a YAML file
+
+    Parameters
+    ----------
+    path : str, optional
+        Path to the configuration file to be loaded. If None, loads the
+        default configuration.
+
+    Returns:
+        A dictionary object
+    """
+    assert isinstance(path, (str, type(None))), 'path must be a string'
```

Comment:
To be consistent, please raise errors instead of using asserts.

Reply:
Thanks for the comments.  I forgot about the asserts.  This has been fixed.